src/main/java/org/apache/commons/lang3/time/DurationFormatUtils.java

```diff
@@ -621,22 +672,23 @@ static boolean containsTokenWithValue(final Token[] tokens, final Object value)
 
         private final Object value;
         private int count;
-        private boolean optional = false;
-
+        private int optionalIndex = -1;
         Token(final Object value) {
             this.value = value;
             this.count = 1;
         }
-        
+
         /**
          * Wraps a token around a value. A value would be something like a 'Y'.
          *
          * @param value to wrap, non-null.
          */
-        Token(final Object value, final boolean optional) {
+        Token(final Object value, final boolean optional, final int optionalIndex) {
```

Comment:
You should javadoc the other parameters and why they are used.

Note: It seems odd to have the other two constructors. These are only used in the tests. If you concur I think it sensible to remove them and add a helper method in the test itself to create these objects using the constructor that your code uses. IIUC the helper method can pass in the desired Object value then call increment a number of times as appropriate.

Reply:
I found the presence of the test-only constructors odd as well. I went ahead and removed them and replaced with a helper method on the test class.